src/charts/heatmap.js

```diff
@@ -0,0 +1,350 @@
+         * @return  {HeatmapData}               Clean data
+         * @private
+         */
+        function cleanData(originalData) {
+            return originalData.reduce((acc, {day, hour, value}) => {
+                return [
+                    ...acc,
+                    {
+                        day: +day,
+                        hour: +hour,
+                        value: +value
+                    }
+                ];
+            }, []);
+        }
+
+        /**
+         * Creates the scales for the heatmap chart
+         * @return void
+         */
+        function buildScales() {
+            colorScale = d3Scale.scaleLinear()
+                .range([colorSchema[0], colorSchema[colorSchema.length - 1]])
+                .domain(d3Array.extent(data, getValue))
+                .interpolate(d3Interpolate.interpolateHcl);
+        }
+
+        /**
+         * Draws the boxes of the heatmap
+         */
+        function drawBoxes() {
+            boxes = svg.select('.chart-group').selectAll('.box').data(data);
+
+            boxes.enter()
+              .append('rect')
+                .classed('box', true)
+                .attr('width', boxSize)
+                .attr('height', boxSize)
+                .attr('x', ({hour}) => hour * boxSize)
+                .attr('y', ({day}) => day * boxSize)
+                .style('opacity', boxInitialOpacity)
+                .style('fill', boxInitialColor)
+                .style('stroke', boxBorderColor)
+                .style('stroke-width', boxBorderSize)
+                .transition()
+                    .duration(animationDuration)
+                    .style('fill', ({value}) => colorScale(value))
+                    .style('opacity', boxFinalOpacity);
+
+            boxes.exit().remove();
+        }
+
+        /**
+         * Draws the day labels
+         */
+        function drawDayLabels() {
+            let dayLabelsGroup = svg.select('.day-labels-group');
+
+            dayLabels = svg.select('.day-labels-group').selectAll('.day-label')
+                .data(daysHuman);
+
+            dayLabels.enter()
+              .append('text')
+                .text((label) => label)
+                .attr('x', 0)
+                .attr('y', (d, i) => i * boxSize)
+                .style('text-anchor', 'start')
+                .style('dominant-baseline', 'central')
+                .attr('class', 'day-label');
+
+            dayLabelsGroup.attr('transform', `translate(-${dayLabelWidth}, ${boxSize / 2})`);
+        }
+
+        /**
+         * Draws the hour labels
+         */
+        function drawHourLabels() {
+            let hourLabelsGroup = svg.select('.hour-labels-group');
+
+            hourLabels = svg.select('.hour-labels-group').selectAll('.hour-label')
+                .data(hoursHuman);
+
+            hourLabels.enter()
+              .append('text')
+                .text((label) => label)
```

Comment:
I think it can be just `.text(label)`

Reply:
Hmmm... `label` is not a variable at this point.
This came from:
  .text((d) => d)